Source/Core/Core/PowerPC/JitArm64/JitArm64_Branch.cpp

```diff
@@ -281,6 +406,18 @@ void JitArm64::bclrx(UGeckoInstruction inst)
   gpr.Flush(conditional ? FlushMode::MaintainState : FlushMode::All, WB);
   fpr.Flush(conditional ? FlushMode::MaintainState : FlushMode::All, ARM64Reg::INVALID_REG);
 
+  if (IsDebuggingEnabled())
+  {
+    BitSet32 gpr_caller_save = {};
+    BitSet32 fpr_caller_save = {};
+    if (conditional)
+    {
+      gpr_caller_save = gpr.GetCallerSavedUsed() & ~BitSet32{DecodeReg(WB), DecodeReg(WC)};
+      fpr_caller_save = fpr.GetCallerSavedUsed();
+    }
```

Comment:
You need to save WA unless js.op->branchIsIdleLoop is true.

I suggest doing it as follows:

```suggestion
    BitSet32 gpr_caller_save = gpr.GetCallerSavedUsed() & ~BitSet32{DecodeReg(WB), DecodeReg(WC)};
    BitSet32 fpr_caller_save = fpr.GetCallerSavedUsed();
    if (js.op->branchIsIdleLoop)
      gpr_caller_save[DecodeReg(WA)] = false;
```

Note that flushing with FlushMode::All flushes all guest registers, not all host registers. So the register cache will still consider WA to be in use until you call Unlock(WA).

Reply:
Isn't it still worth doing less work if conditional is true? Let me know if my latest commits are still missing the mark here.

Reply:
Ah, you mean skipping the work of calling `GetCallerSavedUsed`? That's true, that might be worth skipping. I was thinking more about the source code looking messy. Feel free to do it either way, but if you skip calling GetCallerSavedUsed in the `!conditional` case, make sure to manually set `WA` to saved when `!js.op->branchIsIdleLoop`.
Perhaps, instead of my suggestion of `if (js.op->branchIsIdleLoop) gpr_caller_save[DecodeReg(WA)] = false;`, you could do `gpr_caller_save[DecodeReg(WA)] = !js.op->branchIsIdleLoop;`.

Reply:
Ah, I see now you already implemented it. Your current implementation seems fine.